src/Faithlife.WebRequests/WebServiceRequestSettings.cs

```diff
@@ -87,6 +87,12 @@ public WebServiceRequestSettings()
 		/// </summary>
 		public Func<HttpRequestMessage, IDisposable> StartTrace { get; set; }
 
+		/// <summary>
+		/// A delegate that, if set, is called to retrieve an <see cref="HttpClient"/>.
+		/// If this property is set, then <see cref="CookieManager"/> and <see cref="WebServiceRequestBase.DisableAutoRedirect"/> are ignored.
+		/// </summary>
+		public Func<HttpClient> GetHttpClient { get; set; }
```

Comment:
Is the consuming implementation required/allowed/forbidden to call `Dispose` on the value returned by this method? (It looks like we currently don't call `Dispose` on our `HttpClient` instance.)

Is the providing implementation required to provide a new `HttpClient` each time? Looks like the answer is probably "yes", because we mutate (at least) the `.Timeout` property on it. Sharing instances would be bad.

I like the clarity around usage that the [`IHttpClientFactory.CreateClient`](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.ihttpclientfactory.createclient?view=aspnetcore-2.1) docs give.

Reply:
Presumably the purpose of this is to be able to share the instance. Probably should only set `Timeout` when creating a new `HttpClient` instance and otherwise ignore it like the other properties.

Reply:
I agree; `Timeout` should only be set when creating a new `HttpClient` instance.